src/controls/UserEditInteraction.js

```diff
@@ -0,0 +1,221 @@
+      });
+    });
+    this.modifyInteraction = new Modify({
+      source: this.#editLayer.getSource(),
+    });
+
+    this.modifyInteraction.on('modifystart', event => {
+      const modifiedFeatureIds = event.features
+        .getArray()
+        .map(feature => feature.get('id'))
+        .filter(id => !!id);
+      if (modifiedFeatureIds.length) {
+        for (let i = 0, ii = modifiedFeatureIds.length; i < ii; i++) {
+          const modifiedFeatureId = modifiedFeatureIds[i];
+          if (!this.#modifiedFeatureIDs.includes(modifiedFeatureId)) {
+            this.#modifiedFeatureIDs.push(modifiedFeatureId);
+          }
+        }
+        this.#editSourceLayer.changed();
+      }
+    });
+    this.modifyInteraction.on('modifyend', event => {
+      this.#handleClick(event);
+    });
+    this.drawInteraction.setActive(false);
+    this.modifyInteraction.setActive(false);
+  }
+
+  setMap(map) {
+    const currentMap = this.getMap();
+    if (currentMap) {
+      currentMap.removeLayer(this.#editLayer);
+    }
+    if (!map) {
+      return;
+    }
+    this.#editLayer.setMap(map);
+    super.setMap(map);
+    this.modifyInteraction.setMap(this.#map);
+    this.drawInteraction.setMap(this.#map);
+  }
+
+  drawInteraction;
+
+  modifyInteraction;
+
+  /**
+   * @param {import('ol/MapBrowserEvent.js').default} event
+   */
+  #handleClick(event) {
+    this.#yioMap.dispatchEvent(new PointerEvent('click', event.originalEvent));
+  }
+
+  /**
+   * original style of the content layer
+   */
+  #originalStyle = null;
+
+  setActive(active) {
+    if (!this.modifyInteraction || !this.drawInteraction) {
+      return;
+    }
+    const sourceLayer = getLayerForMapboxSourceLayer(
+      this.#yioMap._getContentLayer(),
+      this.#yioMap.editLayer,
+    );
+    if (active) {
+      this.#editSourceLayer = sourceLayer;
+      this.#originalStyle = sourceLayer.getStyle();
+      sourceLayer.setStyle((feature, resolution) => {
+        if (this.#modifiedFeatureIDs.includes(feature.get('id'))) {
+          return;
+        }
+        return this.#originalStyle(feature, resolution);
+      });
+      this.#setStyle();
+    } else {
+      if (this.#originalStyle) {
+        this.#editSourceLayer.setStyle(this.#originalStyle);
+      }
+      this.#editSourceLayer = null;
+    }
+    this.modifyInteraction.setActive(active);
+    this.drawInteraction.setActive(active);
+    this.#editLayer.getSource().clear();
+    super.setActive(active);
+  }
+
+  /**
+   * sets the style of the editing layer to match the existing layer of the content map.
+   */
+  #setStyle() {
+    this.#editLayer.setStyle((feature, resolution) => {
+      return this.#originalStyle(feature, resolution) || defaultStyles;
+    });
+  }
+
+  /**
+   * @param {import('ol/MapBrowserEvent.js').default} event
+   * @returns {boolean}
+   */
+  handleEvent(event) {
+    let propagateEvent = true;
+
+    if (event.type === 'click') {
+      const map = this.#yioMap._getMap();
```

Comment:
No need to add a `_getMap` method to YioMap, you can get the map on the interaction with `this.getMap()`